clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -577,40 +564,36 @@ raw2trace_t::append_bb_entries(uint tidx, offline_entry_t *in_entry, OUT bool *h
             }
         } else
             prev_instr_was_rep_string = false;
-        buf->size = (ushort)(skip_icache ? 0 : instr_length(dcontext, instr));
+        buf->size = (ushort)(skip_icache ? 0 : instr->length());
         buf->addr = (addr_t)orig_pc;
         ++buf;
         decode_pc = pc;
         // We need to interleave instrs with memrefs.
         // There is no following memref for (instrs_are_separate && !skip_icache).
         if ((!instrs_are_separate || skip_icache) &&
             // Rule out OP_lea.
-            (instr_reads_memory(instr) || instr_writes_memory(instr))) {
-            for (int j = 0; j < instr_num_srcs(instr); j++) {
-                if (opnd_is_memory_reference(instr_get_src(instr, j))) {
-                    std::string error =
-                        append_memref(&buf, tidx, instr, instr_get_src(instr, j), false);
-                    if (error == FAULT_INTERRUPTED_BB) {
-                        truncated = true;
-                        break;
-                    } else if (!error.empty())
-                        return error;
-                }
+            (instr->reads_memory() || instr->writes_memory())) {
+            for (uint j = 0; j < instr->srcs(); j++) {
+                std::string error =
+                    append_memref(&buf, tidx, instr, instr->src_at(j), false);
```

Comment:
You're missing the check for being a memory ref.
UPDATE: see below: srcs and src_at would be much clearer with names like num_mem_srcs and mem_src_at or sthg.

Reply:
done, renamed